### PR TITLE
[receiver/sqlserver] Add more metrics

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

This change adds a query, scraper, and some more metrics for data from the performance counter SQL server table. The query itself is mostly taken from Telegraf's SQL server plugin. This also reuses the existing metric `sqlserver.lock.wait.rate`, so now it will be available both from Windows performance counters, as well as other OSs when directly connecting to the SQL server instance.

**Naming and format feedback on new metrics would be greatly appreciated.**

**Link to tracking Issue:** <Issue number if applicable>
#29865

**Testing:** <Describe what testing was performed and which tests were added.>
Added tests for the query itself, as well as the new scraper.

### CODE REVIEW

File: receiver/sqlserverreceiver/queries.go [hunk 1]
```diff
@@ -62,3 +63,181 @@ func getSQLServerDatabaseIOQuery(instanceName string) string {
...
+	ON (
+		pc.[counter_name] = REPLACE(pc1.[counter_name],' base','')
+		OR pc.[counter_name] = REPLACE(pc1.[counter_name],' base',' (ms)')
+	)
+	AND pc.[object_name] = pc1.[object_name]
+	AND pc.[instance_name] = pc1.[instance_name]
+	AND pc1.[counter_name] LIKE '%base'
+WHERE
+	pc.[counter_name] NOT LIKE '% base'
+{filter_instance_name}
```
Comment (crobert-1) [inline receiver/sqlserverreceiver/queries.go:230]: This allows us to filter based on instance name, replacing the `{filter_instance_name}`. I would have liked to have this indented with the above statement, but this breaks tests directly comparing the contents of two queries when loading the expected query from a file. The file would need to have a blank line with a tab, which my editor automatically deletes, and would look bad on GitHub.

File: receiver/sqlserverreceiver/testdata/perfCounterQueryWithInstanceName.txt [hunk 2]
```diff
@@ -0,0 +1,165 @@
+
```
Comment (crobert-1) [inline receiver/sqlserverreceiver/testdata/perfCounterQueryWithInstanceName.txt:1]: Empty newline is due to formatting of query in go.

File: receiver/sqlserverreceiver/scraper_test.go [hunk 3]
```diff
@@ -47,13 +50,17 @@ func TestSuccessfulScrape(t *testing.T) {
...
+	cfg.MetricsBuilderConfig.Metrics.SqlserverResourcePoolDiskThrottledWriteRate.Enabled = true
+	cfg.MetricsBuilderConfig.Metrics.SqlserverProcessesBlocked.Enabled = true
+
 	scrapers := setupSQLServerScrapers(receivertest.NewNopCreateSettings(), cfg)
 	assert.NotNil(t, scrapers)
 
 	for _, scraper := range scrapers {
 		err := scraper.Start(context.Background(), componenttest.NewNopHost())
 		assert.NoError(t, err)
-		defer func() { assert.NoError(t, scraper.Shutdown(context.Background())) }()
```
Comment (crobert-1) [inline receiver/sqlserverreceiver/scraper_test.go:56]: The anonymous `func()` has to be removed here, as with it, `scraper.Shutdown` is called twice on the second of two scrapers, and not called on the first scraper. (This is how defer works with anonymous funcs by definition, we just need to use them properly for the updated use case.)

**Comment by crobert-1** [comment]
Failing test is frequency of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32715, I've added a reference there.

**Review by djaglowski — COMMENTED** [review]
lgtm

**Review by djaglowski — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**